File: algorithms/0605.can_place_flowers/can_place_flowers.rs
impl Solution {
    pub fn can_place_flowers(flowerbed: Vec<i32>, n: i32) -> bool {
        let mut flowerbed = flowerbed.clone();
        flowerbed.insert(0, 0); flowerbed.push(0);

        for i in 1..flowerbed.len()-1 {
            if flowerbed[i] == 0 && flowerbed[i-1] == 0 && flowerbed[i+1] == 0 {
                flowerbed[i] = -1;
            }
        }
        // println!("flowerbed = {:?}", flowerbed);
        flowerbed.iter().filter(|&&x| x == -1).count() >= n as usize
        // let mut count = 0;
        // for &n in flowerbed.iter() {
        //     if n == -1 {
        //         count += 1;
        //     }
        // }
        // count >= n
    }

}

File: algorithms/0605.can_place_flowers/can_place_flowers.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        int pre = 0;
        flowerbed.push_back(0);
        for (int i = 0; i < flowerbed.size() - 1; ++i) {
            if (flowerbed[i] == 0 && pre == 0 && flowerbed[i+1] == 0) {
                flowerbed[i] = 1;
                n -= 1;
            }
            pre = flowerbed[i];
        }
        return n <= 0;
    }
};